src/I18n/MessagesFileLoader.php

```diff
@@ -157,29 +157,32 @@ public function translationsFolders(): array
         $locale = Locale::parseLocale($this->_locale) + ['region' => null];
 
         $folders = [
-            implode('_', [$locale['language'], $locale['region']]),
             $locale['language'],
+            // gettext compatible paths, see https://www.php.net/manual/en/function.gettext.php
+            $locale['language'] . DIRECTORY_SEPARATOR . 'LC_MESSAGES',
         ];
+        if (!empty($locale['region'])) {
+            $languageRegion = implode('_', [$locale['language'], $locale['region']]);
+            $folders[] = $languageRegion;
+            // gettext compatible paths, see https://www.php.net/manual/en/function.gettext.php
+            $folders[] = $languageRegion . DIRECTORY_SEPARATOR . 'LC_MESSAGES';
```

Comment:
Should we have tests for regional dialects?

Reply:
testTranslationsFoldersGettextCompatible has been added.